addons/point_of_sale/models/pos_order.py

```diff
@@ -1420,51 +1422,161 @@ def get_sale_details(self, date_start=False, date_stop=False, config_ids=False,
                 else:
-                    taxes.setdefault(0, {'name': _('No Taxes'), 'tax_amount':0.0, 'base_amount':0.0})
-                    taxes[0]['base_amount'] += line.price_subtotal_incl
+                    key1 = line.product_id.product_tmpl_id.pos_categ_id.name
+                    key2 = (line.product_id, line.price_unit, line.discount)
+                    refund_done.setdefault(key1, {})
+                    refund_done[key1].setdefault(key2, 0.0)
+                    refund_done[key1][key2] += line.qty
+
+                    if line.tax_ids_after_fiscal_position:
+                        line_taxes = line.tax_ids_after_fiscal_position.sudo().compute_all(line.price_unit * (1-(line.discount or 0.0)/100.0), currency, line.qty, product=line.product_id, partner=line.order_id.partner_id or False)
+                        for tax in line_taxes['taxes']:
+                            refund_taxes.setdefault(tax['id'], {'name': tax['name'], 'tax_amount':0.0, 'base_amount':0.0})
+                            refund_taxes[tax['id']]['tax_amount'] += tax['amount']
+                            refund_taxes[tax['id']]['base_amount'] += tax['base']
+                    else:
+                        refund_taxes.setdefault(0, {'name': _('No Taxes'), 'tax_amount':0.0, 'base_amount':0.0})
+                        refund_taxes[0]['base_amount'] += line.price_subtotal_incl
 
         payment_ids = self.env["pos.payment"].search([('pos_order_id', 'in', orders.ids)]).ids
         if payment_ids:
             self.env.cr.execute("""
-                SELECT COALESCE(method.name->>%s, method.name->>'en_US') as name, sum(amount) total
+                SELECT COALESCE(method.name->>%s, method.name->>'en_US') as name, sum(amount) total, method.is_cash_count as cash
                 FROM pos_payment AS payment,
                      pos_payment_method AS method
                 WHERE payment.payment_method_id = method.id
                     AND payment.id IN %s
-                GROUP BY method.name
+                GROUP BY method.name, method.is_cash_count
             """, (self.env.lang, tuple(payment_ids),))
             payments = self.env.cr.dictfetchall()
         else:
             payments = []
 
-        return {
-            'date_start': date_start,
-            'date_stop': date_stop,
-            'currency_precision': user_currency.decimal_places,
+        config = False
+        session = False
+        if session_ids:
+            session = self.env['pos.session'].browse(session_ids[0])
+        if config_ids:
+            config_id = config_ids[0]
+            config = self.env['pos.config'].search([('id', '=', config_id)])
+        elif session_ids:
+            config_id = session.config_id
+            config = self.env['pos.config'].browse(config_id).id
+
+        cash_payment = False
+        if session:
+            if session.cash_register_balance_end_real:
+                cash_end = session.cash_register_balance_end_real
```

Comment:
The method allows the possibility to get report data from multiple sessions and configs (hence `session_ids` and `config_ids`) but in this implementation, we only care on the first one ?